zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java

```diff
@@ -749,15 +749,25 @@ private static int compareDurationRowTimestamp(DurationRow d1, DurationRow d2) {
     }
 
     /** Returns a map of trace id to timestamp (in microseconds) */
-    public ListenableFuture<Map<Long, Long>> getTraceIdsByDuration(String serviceName, String spanName,
-                                                                   long minDuration, long maxDuration,
-                                                                   long endTs, long startTs, int limit) {
-        int startBucket = durationIndexBucket(startTs);
-        int endBucket = durationIndexBucket(endTs);
+    public ListenableFuture<Map<Long, Long>> getTraceIdsByDuration(String serviceName,
+                                                                   String spanName,
+                                                                   long minDuration,
+                                                                   long maxDuration,
+                                                                   long endTs,
+                                                                   long startTs,
+                                                                   int limit,
+                                                                   int ttl) {
+
+        long oldestData = TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis() - TimeUnit.SECONDS.toMillis(ttl));
+        long safeStartTs = 0 == startTs ? startTs : Math.max(startTs, oldestData);
+        long safeEndTs = 0 == endTs ? endTs : Math.max(endTs, oldestData);
```

Comment:
there's something odd there with the tests and how they sometimes invoke with with empty timestamps. i didn't get my head properly around that, rather just accepted zero as a value that won't get adjusted. i think that's correct but am not entirely sure.